rules/manifest.bzl

```diff
@@ -98,6 +98,27 @@ def _manifest_impl(ctx):
 
     mf["usage_constraints"] = uc
 
+    if ctx.attr.extensions:
+        mf["extensions"] = ctx.attr.extensions
+    else:
+        mf["extensions"] = [
+            "spx_key",
+            "spx_signature",
+            None,
+            None,
+            None,
+            None,
+            None,
+            None,
+            None,
+            None,
+            None,
+            None,
+            None,
+            None,
+            None,
+        ]
```

Comment:
Fair point. [The extension table in `manifest.h`](https://github.com/lowRISC/opentitan/blob/d08268e96afbb793f99fa164a39864ff4d981bea/sw/device/silicon_creator/lib/manifest.h#L410-L412) doesn't make it into [`manifest.rs`](https://github.com/lowRISC/opentitan/blob/master/sw/host/opentitanlib/src/image/manifest.rs) through bindgen, so unless this gets addressed there will be a duplicate source of truth for the layout of this table. Since it's a field in the manifest, I thought it most reasonable to define it here, but if you think it should be baked into opentitantool I can make that change.

Reply:
How is this intended to be used when one actually declares a manifest in bazel?  Suppose I want to declare a bog-standard manifest with no additional extensions (ie: no spx signatures) vs a manifest that does have SPX extensions.

Reply:
If you're declaring the manifest, you can leave the extension table empty in the manifest spec. If you do this, opentitantool will not be able to add the SPX extensions, as the extensions table layout comes from the manifest spec, so opentitantool won't know where in the table to add these extensions.

If you want the flexibility to add the SPX extensions, you can declare the extensions in the extensions table of the manifest spec. Unused extensions are pruned from the table during `image manifest update`.